libcudacxx/include/cuda/std/__random/bernoulli_distribution.h

```diff
@@ -0,0 +1,167 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===////===----------------------------------------------------------------------===//
+
+#ifndef _CUDA_STD___BERNOULLI_DISTRIBUTION_H
+#define _CUDA_STD___BERNOULLI_DISTRIBUTION_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__random/is_valid.h>
+#include <cuda/std/__random/uniform_real_distribution.h>
+#if !_CCCL_COMPILER(NVRTC)
+#  include <ios>
+#endif // !_CCCL_COMPILER(NVRTC)
+
+#include <cuda/std/__cccl/prologue.h>
+
+_CCCL_BEGIN_NAMESPACE_CUDA_STD
+
+class bernoulli_distribution
+{
+public:
+  // types
+  typedef bool result_type;
+
+  class param_type
+  {
+    double __p_;
+
+  public:
+    typedef bernoulli_distribution distribution_type;
+
+    constexpr _CCCL_API explicit param_type(double __p = 0.5)
+        : __p_(__p)
+    {}
+
+    [[nodiscard]] _CCCL_API constexpr double p() const noexcept
+    {
+      return __p_;
+    }
+
+    friend _CCCL_API constexpr bool operator==(const param_type& __x, const param_type& __y) noexcept
+    {
+      return __x.__p_ == __y.__p_;
+    }
+    friend _CCCL_API constexpr bool operator!=(const param_type& __x, const param_type& __y) noexcept
+    {
+      return !(__x == __y);
+    }
```

Comment:
Nitpick: only needed in C++17
```suggestion
#if _CCCL_STD_VER <= 2017
    [[nodiscard]] _CCCL_API friend constexpr bool operator!=(const param_type& __x, const param_type& __y) noexcept
    {
      return !(__x == __y);
    }
#endif // _CCCL_STD_VER <= 2017
```